### ISSUE TITLE
Use of custom alertmanager template 

### ISSUE THREAD

**Issue by herveleclerc** [prometheus-operator/kube-prometheus#784]
Is possible like alertmanager config file (importstr) to use template files created in secret(s) mounted in /etc/alertmanager/*.tmpl or /etc/alertmanager/config/templates.tmpl ?

**Comment by simonpasquier** [comment]
:wave: you can reference and import any file that is mounted the Alertmanager pod: from the secret containing the Alertmanager config, from a secret (listed in `spec.secrets`) or from a configmap (listed in `spec.configMaps`).

You can check https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/alerting.md#manually-managed-secret too.

**Comment by herveleclerc** [comment]
Thx for the quick answer :) Is it possible to do it directly in the main jsonnet file via for example something like that : 
(I don't know the way  to import te template files) 
```
alertmanager+:: {
          config: importstr 'alertmanager-config.yaml',
        },
```

**Comment by simonpasquier** [comment]
Hmm I guess your request is more about [kube-prometheus](https://github.com/prometheus-operator/kube-prometheus#alertmanager-configuration) usage. AFAICT you can't inject "directly" additional items in the config secret. But you should be able to override the generated secret using Jsonnet.

**Comment by Blizter** [comment]
*Note : this an extension of the original question, if it has to be in an issue by itself I'll happily oblige.*

tldr; custom template injected in two steps, first generate a configMap importing external `tmpl` file (not working) , inject generated configMap with `prometheus-operator` api property. 

I am in similar situation where I am trying to inject a custom template file that is separated from the alertmanager configuration. Here is a proposal to do so : 

1. generate a ConfigMap containing the templates (not able to generate one by importing and external `.tmpl` file)
2. Mount configMap inside alertmanager (managed to do so, read following paragraph)

According [while the operator documentation](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md) `configMaps` enables to add a list of configMaps that will be mounted in `/etc/alertmanger/confgimaps/`. Using @kakkoyun approach that removed the use of ksonnet, here is what I came up with inside the main `jsonnet` file, it manages to build without errors yet no configMap is generated :

```jsonnet

local configmap(name, namespace, data) = {
    apiVersion: "v1",
    kind: "ConfigMap",
    metadate : {
        name: name,
        namespace: namespace,
    },
    data: data,
 };

local kp = 
    // different libsonnet imported
  {
      configmap+::·{¬
          'alert-templates':·configmap(¬
          'alert-templates.tmpl',
          $._config.namespace, 
          {"data":·importstr·'foo.tmpl'},
          )
      },
      alertmanager+:{
            spec+:{
                # the important field configmaps:
                configMaps: ['alert-templates',], # goes to etc/alermanager/configmaps
            },
      },
};
```
Please, let me know how if you are able to generate a configmap importing an external template file! Also, I hope this helps you solve you configmap injection issue :smiley: 

Thanks in advance!
Merci d'avance!

**Comment by brancz** [comment]
@Blizter I think this is a fantastic example to be in the docs!

**Comment by Blizter** [comment]
> @Blizter I think this is a fantastic example to be in the docs!

Nice! doing a PR right now.
